operator/controllers/kyma_controller.go

```diff
@@ -504,3 +522,71 @@ func (r *KymaReconciler) GetModules(ctx context.Context, kyma *v1alpha1.Kyma) (p
 
 	return modules, nil
 }
+
+func (r *KymaReconciler) UpdateAndRemoveNotExistsModuleInfo(ctx context.Context, kyma *v1alpha1.Kyma) bool {
+	requireUpdateCondition := false
+	moduleInfoMap := kyma.GetModuleInfoMap()
+	moduleInfos := kyma.GetNotExistsModuleInfos()
+	if len(moduleInfos) == 0 {
+		return false
+	}
+	for i := range moduleInfos {
+		moduleInfo := moduleInfos[i]
+		_, err := r.getModule(ctx, moduleInfo)
```

Comment:
I think this is not well optimized since you do a get request for all modules. Instead you could group them by GVK and then do a list request with a label selector filter, and then map it back to the existing module infos. For now should be okay though

Reply:
completely agree, I think we need a list request and do updating conditions, templateInfos( when it move out of conditions) all at once, currently by sync conditions, it's also get request one by one, here will be a performance issue.